biom/table.py

```diff
@@ -3216,7 +3216,7 @@ def remove_empty(self, axis='whole', inplace=True):
             axes = [axis]
 
         for ax in axes:
-            table.filter(lambda v, i, md: (v > 0).sum(), axis=ax)
+            table.filter(table.ids(axis=ax)[table.sum(axis=ax) > 0], axis=ax)
```

Comment:
I _think_ this should be cast to a `set`, e.g: table.filter(set(table.ids(axis=ax)[table.sum(axis=ax) > 0]), axis=ax)`

Reply:
I might have misremembered here as it looks like a general iterable just works :)

Reply:
👍